src/Expr.cc

```diff
@@ -3903,7 +3897,15 @@ ValPtr RecordCoerceExpr::Fold(Val* v) const
 	if ( same_type(GetType(), Op()->GetType()) )
 		return IntrusivePtr{NewRef{}, v};
 
-	auto val = make_intrusive<RecordVal>(GetType<RecordType>());
+	auto rt = cast_intrusive<RecordType>(GetType());
+	return coerce_to_record(rt, v, map);
+	}
+
+RecordValPtr coerce_to_record(RecordTypePtr rt, Val* v,
+				const std::vector<int>& map)
+	{
+	auto map_size = map.size();
```

Comment:
Unused variable?

Reply:
It's the index for the loop here: https://github.com/zeek/zeek/blob/9a65bb4cd0be45ac8d95206c85516be333ae2001/src/Expr.cc#L3913